src/Microsoft.Identity.Web/AppServicesAuth/AppServicesAuthenticationHandler.cs

```diff
@@ -53,10 +53,10 @@ protected override Task<AuthenticateResult> HandleAuthenticateAsync()
                     ClaimsPrincipal claimsPrincipal = new ClaimsPrincipal(new ClaimsIdentity(
                         jsonWebToken.Claims,
                         idp,
-                        "name", // v1.0
+                        Constants.NameClaim, // v1.0
```

Comment:
That makes me think that we need to check the "ver" claim, and use the right constant (NameClaim or prefered_username, depending on the version.